### ISSUE TITLE
Autofix is corrupting code

### ISSUE THREAD

**Issue by mvorisek** [sindresorhus/eslint-plugin-unicorn#2003]
Steps to reproduce:

checkout https://github.com/fomantic/Fomantic-UI/tree/5e0eede18915f8b048fbb92deec8156f9d659a37

run `eslint -c .eslintrc.js --ext .js . --fix`

actual output:
```
Fomantic-UI\src\definitions\modules\toast.js
  277:148  error  Parsing error: Identifier directly after number

Fomantic-UI\test\meteor\assets.js
  12:55  error  Parsing error: Identifier directly after number

Fomantic-UI\test\meteor\fonts.js
  8:55  error  Parsing error: Identifier directly after number
```

expected output: no parse error (as before eslint autofix)

**Comment by mvorisek** [comment]
the problem is `unicorn/numeric-separators-style`

number separator JS grammar requires `ES2021` even the version in `.eslintrc.js` is defined as `2020`

please advise how to configure the `.eslintrc.js` for `ES2020` grammar or consider removing this rule from the default/recommended ruleset

**Comment by sindresorhus** [comment]
Our rules require the latest JavaScript version. If you don't want that, you can disable the problematic rules.

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/eslint-plugin-unicorn/commit/6aa742d6f661dae852a3bf016973e92a38a29f3e
